types/src/key.rs

```diff
@@ -758,6 +757,15 @@ impl Key {
                     )
                     .map_err(|err| FromStrError::BidAddr(err.to_string()))?;
                     BidAddr::new_delegator_account_addr((validator_bytes, delegator_bytes))
+                } else if tag == BidAddrTag::DelegatedPurse {
```

Comment:
Is there a specific reason this whole thing shouldn't be a `match`?

Reply:
I didn't want to touch existing code, but maybe you're right.